velodyne_driver/src/driver/driver.cc

```diff
@@ -165,11 +165,13 @@ VelodyneDriver::VelodyneDriver(ros::NodeHandle node,
   diag_min_freq_ = diag_freq;
   ROS_INFO("expected frequency: %.3f (Hz)", diag_freq);
 
+  double frequency_tolerance;
+  private_nh.param("frequency_tolerance", frequency_tolerance, 0.1);
```

Comment:
Maybe 'diagnostic_frequency_tolerance' would be better suited, but for the rest I think we can put this PR towards Velodyne main repo. They might have a naming suggestion as well.